src/FSharpVSPowerTools.Logic/EventHandling.fs

```diff
@@ -65,9 +64,13 @@ type DocumentEventListener (events: IEvent<unit> list, delayMillis: uint16, upda
                     startNewTimer()
                     do! awaitPauseAfterChange()
                 protectUpdate() }
-       Async.StartImmediateSafe(computation, tokenSource.Token)
-       // Go ahead and synchronously get the first bit of info for the original rendering
-       protectUpdate()
+       // Go ahead and get the first bit of info for the original rendering
```

Comment:
We don't think we get this part right. The callback function is now run in thread pools. Remember that we had to revert similar changes in this https://github.com/fsprojects/VisualFSharpPowerTools/pull/1141in order to avoid deadlocks.

If you would like to make it async on UI thread, we could do:

```fsharp
    do 
       let computation =
           async { 
            // Go ahead and get the first bit of info for the original rendering
            protectUpdate()
            while true do
                do! Async.AwaitEvent events
                if not skipTimerDelay then
                    startNewTimer()
                    do! awaitPauseAfterChange()
                protectUpdate() }
       Async.StartImmediateSafe (computation, tokenSource.Token)
```